text/0000-dynamic-attributes.md

```diff
@@ -0,0 +1,189 @@
+export default class MyComponent extends LightningElement {
+    childProps = { 
+        name: 'My Child',
+        class: 'my-class',
+        onclick: this.handleClick,
+    };
+    
+    handleClick() {
+        console.log('Clicked')
+    }
+}
+```
+```html
+// x/myComponent.html
+<template>
+    <x-child lwc:spread={childProps}></x-child>
+</template>
+```
+
+The new `lwc:spread` directive passes `childProps` as properties/attributes/event listeners to `x-child`.
+
+## Motivation
+
+In LWC, there is no way to pass props/attributes to a child component when the component author
+doesn't know in advance all the props/attributes that are needed.
+
+One such usecase is `lwc:dynamic` which allows component authors to lazily instantiate components. However,
+the props that we pass to the component can't be dynamic: we need to pass them in the template HTML itself.
+
+This is a problem when the component constructor is changed at runtime: the new constructor
+might need a different set of properties than the earlier one.
+
+We can, of course, not pass any props and once the element renders, query the
+element and set the props programmatically. We lose reactivity this way and also force
+the component to handle `null` props which might not be desirable.
+
+This proposal allows not just `lwc:dynamic` but all other elements to accept an object which is bound as props/attrs/listeners
+at runtime.
+
+### Prior Art
+- React ([JSX Spread Operator](https://reactjs.org/docs/jsx-in-depth.html#spread-attributes))
+- Vue ([`v-bind`](https://v3.vuejs.org/api/directives.html#v-bind))
+- Svelte ([Spread Props](https://svelte.dev/tutorial/spread-props))
+- Lit ([Discussion for support](https://github.com/lit/lit/issues/923), [PR](https://github.com/lit/lit/pull/1960))
+
+## Detailed Design
+
+A new LWC directive, `lwc:spread` will be introduced that can be used to spread the props/attributes/listeners
+to children elements.
+
+### Structure
+It accepts an object with keys as the attribute/property/listener names and values as the corresponding attribute/property/listener values.
+
+This object is merged with props/attrs/listeners passed in the template (essentially `Object.assign`). However, since template
+compiler splits all the props passed in the template into attributes, properties and event listeners, we do the same thing at runtime. We use the same algorithm that the template compiler uses and whatever the [engine does for native custom elements](https://rfcs.lwc.dev/rfcs/lwc/0000-third-party-web-components-support#proposed-solution-2).
+
+There is one difference however: the set of properties passed in the template doesn't change but it changes in `lwc:spread` i.e.
+an attribute can't be present in one render and absent in the next render if it is defined in the template. With
+`lwc:spread` though, it can be present in one render and not be present in the next.
+
+- attrs/listeners are added/removed based on their presence/absence in `lwc:spread`.
```

Comment:
I also disagree with this statement... if you pass `onclick`, we will assign the `onclick` and we will NOT wrap it, or do anything to preserve the context (`this` value) in any ways. If you replace that value with a new `onclick` value, thats up to you, but if you decide to not pass it next time... it doesn't remove the previous value.